### DISCUSSION TITLE
Use fastjson for faster serialization and deserialization

### DISCUSSION THREAD

**Discussion by jdgriffith** [Petersoj/alpaca-java#154] [Ideas]
I'm curious if there is any consideration for other JSON libraries since the amount of messages and subscriptions you can have with market data can be a firehose.

Here is a benchmark project: https://github.com/fabienrenaud/java-json-benchmark?tab=readme-ov-file

I realize you have to take some benchmarks with some nuance, but I think it's a good heuristic. I opted for fastjson when I was using polygon market data so I could process all the incoming trades as quickly as possible. I was generating my own per second aggregates from trades.

I'm interested to hear other's thoughts and if by chance this is a concern for others.

**Comment by Petersoj** [comment]
This is certainly possible, but most users of this library aren't too concerned with the (slower) performance of Gson (de)serialization compared to other libraries. Some [OpenAPI Java generators](https://openapi-generator.tech/docs/generators/java/) use Jackson instead of Gson which may yield some performance improvements, but odds are that there are other performance bottlenecks in one's trading system that show up before JSON (de)serialization. Alpaca isn't meant for HFT as well. 
Just some thoughts.
